app/routers/entitlements.py

```diff
@@ -3,30 +3,48 @@
 from fastapi import APIRouter, HTTPException, status
 from fastapi_pagination.limit_offset import LimitOffsetPage
 
+from app.auth import CurrentSystem
 from app.db.handlers import NotFoundError
-from app.models import EntitlementCreate, EntitlementRead
 from app.pagination import paginate
 from app.repositories import EntitlementRepository
+from app.schemas import EntitlementCreate, EntitlementRead
 
 router = APIRouter()
 
 
 @router.get("/", response_model=LimitOffsetPage[EntitlementRead])
-async def get_entitlements(entitlement_repo: EntitlementRepository):
-    return await paginate(entitlement_repo)
+async def get_entitlements(
+    entitlement_repo: EntitlementRepository,
+    system: CurrentSystem,
+):
+    return await paginate(entitlement_repo, EntitlementRead)
+
+
+@router.post("/", response_model=EntitlementRead, status_code=status.HTTP_201_CREATED)
+async def create_entitlement(
+    data: EntitlementCreate,
+    entitlement_repo: EntitlementRepository,
+    system: CurrentSystem,
+):
+    entitlement = data.to_orm()
+    entitlement.created_by = system
+    entitlement.updated_by = system
+
+    db_entitlement = await entitlement_repo.create(entitlement)
+    return EntitlementRead.from_orm(db_entitlement)
```

Comment:
I would add a generic `except Exception as error` block to manage unexpected errors

Reply:
Probably we should implement a custom fastapi exception handler to handle also such unexpected errors. I will open a task for that.